src/itemlistformaction.cpp

```diff
@@ -1050,6 +1066,25 @@ std::string ItemListFormAction::item2formatted_line(const ItemPtrPosPair& item,
 	return formattedLine;
 }
 
+void ItemListFormAction::goto_item(const std::string& title)
+{
+	const unsigned int curpos = list.get_position();
+	for (unsigned int i = curpos + 1; i < visible_items.size(); ++i) {
+		if (strcasestr(visible_items[i].first->title().c_str(),
+				title.c_str()) != nullptr) {
+			list.set_position(i);
+			return;
+		}
+	}
+	for (unsigned int i = 0; i <= curpos; ++i) {
```

Comment:
With the following config:

```
show-read-articles no
bind-key D goto-title
```

If I enter a feed where all the articles are read (so the list is empty), press D and enter anything (I typed "Newsboat"), Newsboat segfaults because `curpos` and `i` are both zero, so the loop uses non-existent first element of `visible_items` and eventually dereferences a NULL pointer.

I understand that this code is a copy of a similar code in `FeedListFormAction`, but I'm afraid it's on you to fix it anyway ._. The same bug exists in `FeedListFormAction`, of course (swap `show-read-articles` with `show-read-feeds`); please fix it there as well.

Reply:
Good catch, I fixed it by skipping the whole function if the list is empty: https://github.com/newsboat/newsboat/pull/1249/commits/701cd6dd4f181f4116ed065ab904470fcd416385.